lib/net/ssh/authentication/methods/publickey.rb

```diff
@@ -27,41 +27,43 @@ def authenticate(next_service, username, password=nil)
 
           # Builds a packet that contains the request formatted for sending
           # a public-key request to the server.
-          def build_request(pub_key, username, next_service, has_sig)
+          def build_request(pub_key, username, next_service, alg, has_sig)
             blob = Net::SSH::Buffer.new
             blob.write_key pub_key
 
             userauth_request(username, next_service, "publickey", has_sig,
-              pub_key.ssh_type, blob.to_s)
+              alg, blob.to_s)
           end
 
           # Builds and sends a request formatted for a public-key
           # authentication request.
-          def send_request(pub_key, username, next_service, signature=nil)
-            msg = build_request(pub_key, username, next_service, !signature.nil?)
+          def send_request(pub_key, username, next_service, alg, signature=nil)
+            msg = build_request(pub_key, username, next_service, alg,
+              !signature.nil?)
             msg.write_string(signature) if signature
             send_message(msg)
           end
 
           # Attempts to perform public-key authentication for the given
           # username, with the given identity (public key). Returns +true+ if
           # successful, or +false+ otherwise.
-          def authenticate_with(identity, next_service, username)
+          def authenticate_with_2(identity, next_service, username, alg, salg)
             debug { "trying publickey (#{identity.fingerprint})" }
-            send_request(identity, username, next_service)
+            send_request(identity, username, next_service, alg)
```

Comment:
This should be correct as it is, because value of ssh_type (not ssh_signature_type) is passed here, which preserves existing behaviour. See more here:
https://github.com/net-ssh/net-ssh/pull/811#discussion_r629628505